packages/bento-design-system/src/Table/Table.tsx

```diff
@@ -77,15 +78,18 @@ type SortingProps<C extends ReadonlyArray<ColumnType<string, {}, any>>> =
 type Props<C extends ReadonlyArray<ColumnType<string, {}, any>>> = {
   columns: C;
   data: ReadonlyArray<RowType<C>>;
-  groupBy?: C[number]["accessor"];
   noResultsTitle?: LocalizedString;
   noResultsDescription?: LocalizedString;
   noResultsFeedbackSize?: FeedbackProps["size"];
   initialSorting?: Array<SortingRule<C>>;
   stickyHeaders?: boolean;
   height?: { custom: string | number };
   onRowPress?: (row: Row<RowType<C>>) => void;
-} & SortingProps<C>;
+} & (
+  | { groupBy?: C[number]["accessor"]; virtualizeRows?: never }
+  | { groupBy?: never; virtualizeRows?: { estimateRowHeight: (index: number) => number } }
```

Comment:
mutually exclusive for now as I didn't investigate how to support virtualizing tables with `groupBy` enabled